express/src/lib/publicKeyUtils.ts

```diff
@@ -6,17 +6,29 @@ export default function getAuthApiCompliantPublicKey(publicKey: string): string
     publicKey = publicKey.trim();
     // assume PEM with headers
     try {
-        createPublicKey({key: publicKey, format: "pem"});
-        return publicKey.replace(BEGIN, "").replace(END, "").replace(/\n/g, "").replace(/\r/g, "");
+        return makeAuthCompliant(publicKey);
     } catch (ignored) {}
-    // assume they didn't supply headers
-    const publicKeyWithHeaders = `${BEGIN}\n${publicKey}\n${END}\n`;
 
     try {
-        createPublicKey({key: publicKeyWithHeaders, format: "pem"});
-        return publicKeyWithHeaders.replace(BEGIN, "").replace(END, "").replace(/\n/g, "").replace(/\r/g, "");
+        // assume they didn't supply headers
+        const publicKeyWithHeaders = `${BEGIN}\n${publicKey}\n${END}\n`;
+        return makeAuthCompliant(publicKeyWithHeaders);
     } catch (err) {
         console.error(err);
         throw new Error(`Failed to convert\n${publicKey}\n`);
     }
 }
+
+function makeAuthCompliant(publicKey: string): string {
+    const key = createPublicKey({key: publicKey, format: "pem"});
+    return (
+        key.export({
+            format: "pem",
+            type: "spki"
+        }) as string
```

Comment:
Better to use the `toString()` method on the exported object and let node handle the type conversion if needed, and also make typescript happy knowing the thing will always be a stirng wihtout turning off the type checking.

The methods can be chained as well
```
return createPublicKey({key: publicKey, format: "pem"})
        .export({format: "pem", type: "spki"})
        .toString()
        .replace(BEGIN, "")
        ........
```